internal/generate/types.go

```diff
@@ -753,6 +754,15 @@ func createOneOf(s *openapi3.Schema, name, typeName string) ([]TypeTemplate, []E
 
 				enumFieldName = strcase.ToCamel(p.Value.Enum[0].(string))
 			}
+
+			// Enums can appear in a valid OpenAPI spec as a OneOf without necessarily
+			// being identified as such. If we find an object with a single property
+			// nested inside a OneOf we will assume this is an enum and modify the name of
+			// the struct that will be created out of this object.
+			// e.g. https://github.com/oxidecomputer/omicron/blob/158c0b205f23772dc6c4c97633fd1769cc0e00d4/openapi/nexus.json#L18637-L18682
+			if len(keys) == 1 && p.Value.Enum == nil {
+				enumFieldName = propertyName
+			}
```

Comment:
This code handles the case for https://github.com/oxidecomputer/oxide.go/pull/284/files#diff-841a2b1b5f072b633551cac9c34c9732a722b415e297a1530b92d5988962062eR3728-R3750 Without handling, the result was that they all were named `InterfaceNum` because they aren't being explicitly marked as an enum, but used as one, and OpenAPI still sees this as a valid spec

Reply:
Makes sense to me!